lib/puppet/util/windows/adsi.rb

```diff
@@ -196,10 +196,19 @@ def self.logon(name, password)
       Puppet::Util::Windows::User.password_is?(name, password)
     end
 
-    def [](attribute)
+    def get_attribute(attribute)
       # Setting WIN32OLE.codepage in the microsoft_windows feature ensures
       # values are returned as UTF-8
-      native_user.Get(attribute)
+      begin
+        native_user.Get(attribute)
+      rescue WIN32OLERuntimeError => e
+        # This OLE error code indicates the property can't be found in the cache
+        raise e unless e.message =~ /8000500D/m
```

Comment:
I think it might be a good idea to enumerate all of the available properties for the IADsUser -- they should be here https://docs.microsoft.com/en-us/windows/desktop/adsi/iadsuser-property-methods and https://docs.microsoft.com/en-us/windows/desktop/adsi/unsupported-iadsuser-properties and validate that the attribute matches it. Reason is b/c this code would also return nil for a property that can never exist in the property cache, not just one that (possibly) can. Also, any reason for the separate method? I think `[]` read fine.